src/main/webapp/src/app/recommendation/recommendation-form/recommendation-form.component.ts

```diff
@@ -34,13 +37,71 @@ export class RecommendationFormComponent implements OnInit{
     })
 
   }
+
+  ngDoCheck(){
+    if(this.hiddenGems !== this.appComponent.hiddenGems){
+      this.hiddenGems = this.appComponent.hiddenGems;
+    }
+  }
+
   // Submit user's preference form
   // Send hidden gem recommendation as output
-  onSubmit(): void {
-    this.hiddenGemService.findHiddenGemRecommendation(this.preferenceForm.value)
-      .subscribe(hiddenGems => {
-        this.hiddenGemRecommendation.emit(hiddenGems);
-    })
+  async onSubmit() {
+    // filter hidden gems by preference form criteria.
+    let result = this.filterGems(this.hiddenGems, this.preferenceForm);
```

Comment:
Maybe display an error message if they are not results matching the user's preferences? Like "There are no hidden gems matching your preferences" displayed on the screen if result is empty?

Reply:
This error will only be triggered for an empty list of hidden gems. Still added in the error check :)